src/Workspaces/Remote/ServiceHub/Host/ChecksumSynchronizer.cs

```diff
@@ -46,81 +44,59 @@ public async ValueTask SynchronizeSolutionAssetsAsync(Checksum solutionChecksum,
                     checksums.Remove(solutionChecksumObject.Checksum);
                     await _assetProvider.SynchronizeAssetsAsync(checksums, cancellationToken).ConfigureAwait(false);
                 }
+            }
 
-                // third and last get direct children for all projects and documents in the solution 
-                await SynchronizeProjectAssets_NoLockAsync(solutionChecksumObject.Projects, cancellationToken).ConfigureAwait(false);
+            // third and last get direct children for all projects and documents in the solution 
+            foreach (var project in solutionChecksumObject.Projects)
```

Comment:
lifted the loop *out* of the SynchronizeProjectAssets_NoLockAsync helper.  this makes it easy to sync just a project at a time.  note: the next PR ensures that when we go to sync a particular project that the other side can do that work efficiently (without having to scan the entire solution for the data it needs).